### PR TITLE
fix bug when migrate repository 500 when repo is existed

### LINKED ISSUE #4140: If there's a repo with the same name, the migration throws a 500 error
- Gitea version (or commit ref): 1.1.0+1336-gd62bc336
- Git version: 2.15.2-r0
- Operating system: Alpine Linux 3.7 (Docker)
- Database (use `[x]`):
  - [X] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X] Not relevant
- Log gist:

`docker logs gitea`
```
[Macaron] 2018-06-05 18:26:31: Started GET / for 192.168.5.5
[Macaron] 2018-06-05 18:26:32: Completed GET / 200 OK in 69.328709ms
[Macaron] 2018-06-05 18:26:32: Started GET /api/v1/repos/search?uid=1&q=&limit=15&mode= for 192.168.5.5
[Macaron] 2018-06-05 18:26:32: Completed GET /api/v1/repos/search?uid=1&q=&limit=15&mode= 200 OK in 17.317737ms
[Macaron] 2018-06-05 18:26:34: Started GET /repo/migrate for 192.168.5.5
[Macaron] 2018-06-05 18:26:34: Completed GET /repo/migrate 200 OK in 23.573387ms
[Macaron] 2018-06-05 18:26:43: Started POST /repo/migrate for 192.168.5.5
[Macaron] 2018-06-05 18:26:43: Completed POST /repo/migrate 404 Not Found in 27.013216ms
```
`gitea.logs` has no relevant logs

## Description
When you try to migrate a repository and there's already a repository with the name you set, it throws a 500 error

## Screenshots
captura de 2018-06-05 20-40-46

I'll try to solve it, but I don't think I'll be able do it

### PR DESCRIPTION
**Pull request by lunny**
will fix #4140

### CODE REVIEW

**Comment by NefixEstrada** [comment]
Hey! First of all thanks for fixing the bug!

I see you are sending back a 422 HTTP Code. Shouldn't a 409 be more appropiate?

Thanks,
Néfix Estrada

**Comment by lunny** [comment]
@NefixEstrada done.

File: routers/repo/repo.go [hunk 1]
```diff
@@ -256,6 +256,11 @@ func MigratePost(ctx *context.Context, form auth.MigrateRepoForm) {
 		return
 	}
 
+	if models.IsErrRepoAlreadyExist(err) {
+		ctx.RenderWithErr(ctx.Tr("repo.migrate.repo_existed"), tplMigrate, &form)
```
Comment (lafriks) [inline routers/repo/repo.go]: Why not use already existing error message?
```suggestion
		ctx.RenderWithErr(ctx.Tr("form.repo_name_been_taken"), tplMigrate, &form)
```

File: routers/api/v1/repo/repo.go [hunk 2]
```diff
@@ -408,6 +408,11 @@ func Migrate(ctx *context.APIContext, form auth.MigrateRepoForm) {
 		RemoteAddr:  remoteAddr,
 	})
 	if err != nil {
+		if models.IsErrRepoAlreadyExist(err) {
+			ctx.Error(409, "", "The repository with the same name has existed.")
```
Comment (lafriks) [inline routers/api/v1/repo/repo.go]:
```suggestion
			ctx.Error(409, "", "The repository with the same name already exists.")
```

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by lafriks — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
@lunny please backport 😃

**Pull request merged**